### PR TITLE
Change the semantics of Drain to drop any elements remaining when the iterator is dropped

### PR DESCRIPTION
**Pull request by sdleffler**
:)

### CODE REVIEW

File: src/drain.rs [hunk 1]
```diff
@@ -1,4 +1,5 @@
 use std::iter::{ExactSizeIterator, FusedIterator};
+use std::mem::drop;
```
Comment (LPGhatguy) [inline src/drain.rs:2]: `drop` is in the prelude, we need not import it

**Review by LPGhatguy — CHANGES_REQUESTED** [review]
One small change. Do we want to update `clear` to rely on automatic dropping instead of using `.for_each(drop)`?

File: src/drain.rs (hunk 1 above)
Comment (sdleffler) [reply src/drain.rs:2]: Oh oops. I'll remove that.

I think there's a good argument to use `.for_each(drop)` just to be super clear about what's happening. What do you think?

Comment (LPGhatguy) [reply src/drain.rs]: I don't mind either way, we can always revisit that later!

**Review by LPGhatguy — APPROVED** [review]

**Pull request merged**